pollen_data_gen/pollen_data_gen/simple.py

```diff
@@ -1,11 +1,97 @@
+# import sys
 import json
-import dataclasses
-from typing import Dict, Union, Optional, Any
+from typing import Dict, Union, Optional, Any, List, Sequence, TextIO
 from json import JSONEncoder
 from mygfa import mygfa
+from . import depth
 
 
-SimpleType = Optional[Dict[str, Union[bool, str, int]]]
+SimpleType = Optional[
+    Union[str, Dict[str, Sequence[object]], List[int], List[Union[int, str]]]
+]
+
+char_to_number = {"A": 1, "T": 2, "G": 3, "C": 4, "N": 5}
+number_to_char = {v: k for k, v in char_to_number.items()}
+
+
+def strand_to_number_list(strand: mygfa.Strand) -> List[int]:
+    """Converts a strand to a list of numbers following the mapping above.
+    For instance, "AGGA" is converted to [1,3,3,1].
+    """
+    return [char_to_number[c] for c in str(strand)]
+
+
+def number_list_to_strand(numbers: List[str]) -> mygfa.Strand:
+    """Converts a list of numbers to a strand following the mapping above.
+    For instance, [1,3,3,1] is converted to "AGGA"."""
+    return mygfa.Strand("".join([number_to_char[int(number)] for number in numbers]))
+
+
+def path_seq_to_number_list(path: str) -> List[int]:
+    """Converts a path's segment sequence into a list of numbers.
+    Every + becomes 0 and - becomes 1.
+    For instance, "1+,2-,14+" is converted to [1,0,2,1,14,0].
+    The 1 at the 4th cell will not be confused for a node called "1" because
+    it is at an even index.
+    It's ugly but it works for now...
+    """
+    ans = []
+    for chunk in path.split(","):
+        num, orient = chunk[:-1], chunk[-1]
+        ans.append(int(num))
+        if orient == "+":
+            ans.append(0)
+        else:
+            ans.append(1)
+
+    return ans
```

Comment:
This is not actionable for now, but for future iterations of the hardware, we might want to consider splitting this into two arrays (memories): one for the indices and one for the directions. This would save space because the index memory would be $`\log_2 \text{max\_steps}`$ bits per element and the direction memory would be 1 bit per element.

Reply:
Ah nice! I've stuck a comment in there for now, so we don't forget.